canonical/edgex-checkbox-provider
# Add connection test for kuiper

- test Kuiper 1.4.2 new feature, re-use edgex message bus as sink's action
- update rules' format and naming, to improve error logs

## Test guideline
1. build edgexfoundry snap with kuiper version 1.4.2
```
git clone https://github.com/MonicaisHer/edgex-go.git
cd edgex-go
git checkout EDGEX-172-update-kuiper-to-1-4-x
snapcraft
```
2. test kuiper with local edgexfoundry snap
```
cd edgex-checkbox-provider/data/latest
sudo ./run-all-tests-locally.sh -t test-rules-engine.sh -s <.../edgex-go/edgexfoundry_2.0.1-dev.71_amd64.snap> -v
```
Expect:
```
running single test: test-rules-engine.sh ...	PASSED
snap "edgexfoundry" is not installed
edgexfoundry 2.0.1-dev.71 installed
waiting for all services to come online. Current retry count: 1/300
waiting for all services to come online. Current retry count: 2/300
waiting for all services to come online. Current retry count: 3/300
all services up
waiting for device-virtual produce readings, current retry count: 1/60
waiting for device-virtual produce readings, current retry count: 2/60
waiting for device-virtual produce readings, current retry count: 3/60
waiting for device-virtual produce readings, current retry count: 4/60
waiting for device-virtual produce readings, current retry count: 5/60
waiting for device-virtual produce readings, current retry count: 6/60
waiting for device-virtual produce readings, current retry count: 7/60
waiting for device-virtual produce readings, current retry count: 8/60
waiting for device-virtual produce readings, current retry count: 9/60
waiting for device-virtual produce readings, current retry count: 10/60
waiting for device-virtual produce readings, current retry count: 11/60
device-virtual is producing readings now
check port 59720 status open services timed out, current retry count: 1/300
time="2022-01-25 15:12:02" level=info msg="Specified Kuiper base folder at location /var/snap/edgexfoundry/x1/kuiper.\n" file="conf/path.go:100"
time="2022-01-25 15:12:02" level=info msg="Specified Kuiper base folder at location /var/snap/edgexfoundry/x1/kuiper.\n" file="conf/path.go:100"
time="2022-01-25 15:12:03" level=info msg="Specified Kuiper base folder at location /var/snap/edgexfoundry/x1/kuiper.\n" file="conf/path.go:100"
time="2022-01-25 15:12:03" level=info msg="Specified Kuiper base folder at location /var/snap/edgexfoundry/x1/kuiper.\n" file="conf/path.go:100"
time="2022-01-25 15:12:03" level=info msg="Specified Kuiper base folder at location /var/snap/edgexfoundry/x1/kuiper.\n" file="conf/path.go:100"
time="2022-01-25 15:12:03" level=info msg="Specified Kuiper base folder at location /var/snap/edgexfoundry/x1/kuiper.\n" file="conf/path.go:100"
time="2022-01-25 15:12:03" level=info msg="Specified Kuiper base folder at location /var/snap/edgexfoundry/x1/kuiper.\n" file="conf/path.go:100"
waiting for readings come into stream, current retry count: 1/60
time="2022-01-25 15:12:04" level=info msg="Specified Kuiper base folder at location /var/snap/edgexfoundry/x1/kuiper.\n" file="conf/path.go:100"
readings come into stream now
time="2022-01-25 15:12:04" level=info msg="Specified Kuiper base folder at location /var/snap/edgexfoundry/x1/kuiper.\n" file="conf/path.go:100"
time="2022-01-25 15:12:04" level=info msg="Specified Kuiper base folder at location /var/snap/edgexfoundry/x1/kuiper.\n" file="conf/path.go:100"
time="2022-01-25 15:12:04" level=info msg="Specified Kuiper base folder at location /var/snap/edgexfoundry/x1/kuiper.\n" file="conf/path.go:100"
time="2022-01-25 15:12:04" level=info msg="Specified Kuiper base folder at location /var/snap/edgexfoundry/x1/kuiper.\n" file="conf/path.go:100"
time="2022-01-25 15:12:05" level=info msg="Specified Kuiper base folder at location /var/snap/edgexfoundry/x1/kuiper.\n" file="conf/path.go:100"
time="2022-01-25 15:12:05" level=info msg="Specified Kuiper base folder at location /var/snap/edgexfoundry/x1/kuiper.\n" file="conf/path.go:100"
time="2022-01-25 15:12:05" level=info msg="Specified Kuiper base folder at location /var/snap/edgexfoundry/x1/kuiper.\n" file="conf/path.go:100"
time="2022-01-25 15:12:05" level=info msg="Specified Kuiper base folder at location /var/snap/edgexfoundry/x1/kuiper.\n" file="conf/path.go:100"
time="2022-01-25 15:12:05" level=info msg="Specified Kuiper base folder at location /var/snap/edgexfoundry/x1/kuiper.\n" file="conf/path.go:100"
edgexfoundry removed
```
